agent/metrics/metrics_engine.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
should we have a dummy metric engine for platform other than linux so that customers won't try to use it in other platforms and fail?

Reply:
I removed the PrometheusMetricsEnabled env var from the config.go file and placed it under platformOverrides() function in config_unix.go. This will allow only unix Agents to set the environment variable successfully and therefore setup metrics.